MyApplication/app/src/main/java/com/korbkenny/recyclerview_lab/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,53 @@
+package com.korbkenny.recyclerview_lab;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by KorbBookProReturns on 10/18/16.
+ */
+
+public class RecyclerViewAdapter extends RecyclerView.Adapter<CustomViewHolder> {
+    List<ObjectItem> mObjectList;
+
+    @Override
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View parentView = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_item, parent, false);
+        CustomViewHolder viewHolder = new CustomViewHolder(parentView);
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(CustomViewHolder holder, int position) {
+
+        View.OnClickListener onClickListener = new View.OnClickListener() {
```

Comment:
You need to separate the concepts of what you do when clicking on the item from the code used for updating the view itself. What goes in the Click Listener is just code for setting the boolean for the checkbox to either true or false, notifying the adapter that it needs to update, and then .

Outside of the click listener (after the ending curly brace) you put the code that accesses the view holder to set up each of the views. This includes looking at the current object to determine color, title, status of the checkbox, etc. and using those values to set properties for the views.

Basically code in the click listener updates your data (the objects), code outside of it directly updates your view.